inference-engine/src/mkldnn_plugin/nodes/one_hot.cpp

```diff
@@ -36,38 +41,41 @@ class OneHotImpl: public ExtLayerBase {
 
             // check a precision of the input tensor
             input_precision = layer->insData[0].lock()->getTensorDesc().getPrecision();
-            if (input_precision == Precision::BF16)
-                input_precision = Precision::FP32;
-            if (input_precision != Precision::I32 && input_precision != Precision::FP32) {
-                THROW_IE_EXCEPTION << layer->name << " Incorrect input precision for the input. Only I32 and FP32 are supported!";
+            if (!one_of(input_precision, Precision::I32, Precision::FP32, Precision::BF16)) {
+                THROW_IE_EXCEPTION << layer->name << " Incorrect input precision for the input. Only I32, FP32 and BF16 are supported!";
+            }
+            output_precision = layer->outData[0]->getTensorDesc().getPrecision();
+            if (!one_of(output_precision, Precision::FP32, Precision::BF16)) {
+                THROW_IE_EXCEPTION << layer->name << " Incorrect precision for the output. Only FP32 and BF16 are supported!";
             }
 
-            addConfig(layer, { DataConfigurator(ConfLayout::PLN) }, { DataConfigurator(ConfLayout::PLN) });
+            addConfig(layer, { DataConfigurator(ConfLayout::PLN, input_precision) }, { DataConfigurator(ConfLayout::PLN, output_precision) });
         } catch (InferenceEngine::details::InferenceEngineException &ex) {
             errorMsg = ex.what();
         }
     }
 
     StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept override {
-        switch (input_precision) {
-            case Precision::FP32:
-                one_hot<float>(inputs[0], outputs[0]);
-                break;
-            case Precision::I32:
-                one_hot<int>(inputs[0], outputs[0]);
-                break;
-            default:
-                return GENERAL_ERROR;
+        OneHotContext ctx = {this, inputs[0], outputs[0], false};
+        OV_SWITCH(MKLDNNPlugin, OneHotExecute, ctx, std::tie(input_precision, output_precision),
```

Comment:
As we discussed offline: using floating point precisions for indices input is not valid. Should be I32 only.

Reply:
Fixed in 8a2d1b3. Also I8 output precision support was added.